vm/systemSmartContracts/staking.go

```diff
@@ -1353,6 +1381,13 @@ func (s *stakingSC) getWaitingListIndex(args *vmcommon.ContractCallInput) vmcomm
 }
 
 func (s *stakingSC) getWaitingListSize(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if s.flagStakingV4.IsSet() {
+		s.eei.AddReturnMessage(vm.ErrWaitingListDisabled.Error())
+		s.eei.Finish([]byte{0})
```

Comment:
if you have return message than return vmcommon.UserError, and do not add to Finish anything. as it is ignored

Reply:
Thanks for heads up, refactored it